### PR TITLE
Add HtmlPdfDownloadButton panel for YDataProfiler view and fixes analyses button

### PR DESCRIPTION
**Pull request by ahuang11**
Supersedes https://github.com/holoviz/lumen/pull/720

I was initially testing JsPDF, but I discovered it works without it :o

https://github.com/user-attachments/assets/83a1194b-b556-49f9-8687-b6ae18a8a7e4

[Profile Report 2.pdf](https://github.com/user-attachments/files/17468019/Profile.Report.2.pdf)

### CODE REVIEW

**Review by philippjfr — APPROVED** [review]

File: lumen/ai/assistant.py [hunk 1]
```diff
@@ -405,7 +408,7 @@ async def _get_agent(self, messages: list | str):
                 step.stream(f"Assistant decided the {agent_name!r} will provide {', '.join(deps)}.")
                 self._current_agent.object = f"## **Current Agent**: {agent_name}"
                 custom_messages = messages.copy()
-                if isinstance(subagent, SQLAgent):
+                if isinstance(subagent, (TableAgent | SQLAgent)):
```
Comment (philippjfr) [inline lumen/ai/assistant.py:411]: What's the benefit over the simpler

```suggestion
                if isinstance(subagent, (TableAgent. SQLAgent)):
```

Comment (ahuang11) [reply lumen/ai/assistant.py:411]: Ruff complains about the simpler

File: lumen/panel.py [hunk 2]
```diff
@@ -155,3 +156,38 @@ def _button_click(self, event=None):
...
+    _esm = """
+    export function render({ model }) {
+      const button = document.createElement('button');
+      button.textContent = 'Download PDF';
+      button.addEventListener('click', () => {
+        const printWindow = window.open('', '_blank');
+        printWindow.document.write(`
+          <html>
+            <head>
+              <title>Profile Report</title>
```
Comment (philippjfr) [inline lumen/panel.py:174]: We will likely reuse this. No reason to hardcode this.

**Comment by philippjfr** [comment]
Would suggest styling that button to be a little nicer and probably position it to the top right of the output (using absolute or relative positioning).

File: lumen/panel.py (hunk 2 above)
Comment (ahuang11) [inline lumen/panel.py:174]:
```suggestion
```

File: lumen/ai/assistant.py (hunk 1 above)
Comment (philippjfr) [reply lumen/ai/assistant.py:411]: Weird @hoxbro is this now preferred syntax?

Comment (ahuang11) [reply lumen/ai/assistant.py:411]:
```
lumen/ai/assistant.py:411:20: UP038 Use `X | Y` in `isinstance` call instead of `(X, Y)`
```

Comment (hoxbro) [reply lumen/ai/assistant.py:411]: Disable that rule. It is slower than tuple per Andrews link.

**Comment by ahuang11** [comment]
image

image

**Pull request merged**